docs/deployment/gdal_config.md

```diff
@@ -0,0 +1,129 @@
+an initial metadata section that describes the location (byte range) within the
+file of each internal tile. The more internal tiles the COG has, the more data
+the header needs to contain.
+
+GDAL needs to read the entire header before it can read any other portion of the
+file. By default GDAL reads the first 16KB of the file, then if that doesn't
+contain the entire metadata, it makes one more request for the rest of the
+metadata.
+
+In environments where latency is relatively high (at least compared to
+bandwidth), such as AWS S3, it may be beneficial to increase this value
+depending on the data you expect to read.
+
+There isn't currently a way to get the number of header bytes using GDAL, but
+alternative GeoTIFF readers such as [`aiocogeo`][aiocogeo] can. Using its cli
+you can find the image's header size:
+
+[aiocogeo]: https://github.com/geospatial-jeff/aiocogeo
+
+```
+> export AWS_REQUEST_PAYER="requester"
+> aiocogeo info s3://usgs-landsat/collection02/level-2/standard/oli-tirs/2020/072/076/LC08_L2SR_072076_20201203_20201218_02_T2/LC08_L2SR_072076_20201203_20201218_02_T2_SR_B1.TIF
+
+          PROFILE
+            ...
+            Header size:      32770
+```
+
+It's wise to inspect the header sizes of your data sources, and set
+`GDAL_INGESTED_BYTES_AT_OPEN` appropriately. Beware, however, that the given
+number of bytes will be read for every image, so you don't want to make the
+value too large.
+
+#### `GDAL_DISABLE_READDIR_ON_OPEN`
+
+This is a very important setting to control the number of requests GDAL makes.
+
+This setting has two options: `FALSE` and `EMPTY_DIR`. `FALSE` (the default)
+causes GDAL to try to establish a list of all the available files in the
+directory. `EMPTY_DIR` tells GDAL to imagine that the directory is empty except
+for the requested file.
+
+When reading datasets with necessary external sidecar files, it's imperative to
+set `FALSE`. For example, the `landsat-pds` bucket on AWS S3 contains GeoTIFF
+images where overviews are in external `.ovr` files. If set to `EMPTY_DIR`, GDAL
+won't find the `.ovr` files.
+
+However, in all other cases, it's much better to set `EMPTY_DIR` because this
+prevents GDAL from making a `LIST` request.
+
+This setting also has cost implications for reading data from requester-pays
+buckets. When set to `FALSE`, GDAL makes a `LIST` request every time it opens a
+file. Since `LIST` requests are much more expensive than `GET` requests, this
+can bring unexpected costs.
+
+#### `GDAL_HTTP_MERGE_CONSECUTIVE_RANGES`
+
+When set to `YES`, this tells GDAL to merge adjacent range requests. Instead of
+making two requests for byte ranges `1-5` and `6-10`, it would make a single
+request for `1-10`. This should always be set to `YES`.
+
+#### `GDAL_CACHEMAX`
+
+#### `GDAL_DATA`
+
+#### `PROJ_LIB`
+
+#### `VSI_CACHE`
+
+Setting this to `TRUE` enables GDAL to use an internal caching mechanism. It's
+strongly recommended to set this to `TRUE`.
+
+#### `VSI_CACHE_SIZE`
```

Comment:
should mention that this is a per-file handle.  Ex. if you open a VRT with 10 files and your VSI_CACHE_SIZE is 10 bytes, the total cache memory usage would be 100 bytes.